src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateLetterPrintTimesTest.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.hmcts.reform.sendletter.controllers;
+
+import com.microsoft.azure.servicebus.IQueueClient;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.sendletter.data.LetterRepository;
+import uk.gov.hmcts.reform.sendletter.domain.LetterStatus;
+import uk.gov.hmcts.reform.sendletter.model.DbLetter;
+import uk.gov.hmcts.reform.sendletter.model.Letter;
+import uk.gov.hmcts.reform.sendletter.queue.QueueClientSupplier;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class UpdateLetterPrintTimesTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private LetterRepository letterRepository;
+
+    @MockBean
+    private AuthTokenValidator tokenValidator;
+
+    @MockBean
+    private QueueClientSupplier queueClientSupplier;
+
+    @Mock
+    private IQueueClient queueClient;
+
+    @Value("${status-update-service-name}")
+    private String serviceName;
```

Comment:
perhaps `serviceAllowedToMakeUpdates` for clarity in tests?

Reply:
W ill it be sufficient if I add a dockblock instead? Making vars so long seems a bit unnecessary. But docblock is useful